Get data regarding frame count as source, target, use in metaphors, etc.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../metanet/data/trimmed/Metaphor.csv")


In [3]:
df

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
0,Metaphor_POWER_IS_A_POSSESSED_OBJECT,['Frame_Possession'],['Frame_Power'],"['Example_4540', 'Example_4542', 'Example_4541...",['this metaphor is similar to CONTROL IS A POS...,['Composed/complex'],"['Entailment_196', 'Entailment_195', 'Entailme...",['POWER IS A POSSESSED OBJECT'],['Metaphor_family_Possession_metaphors'],['POWER IS A POSSESSED OBJECT'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Metaphor_GOVERNMENT_IS_A_SLAVE_MASTER,['Frame_Work_scenario'],['Frame_Government'],"['Example_2625', 'Example_2621', 'Example_2626...",NaN,NaN,NaN,['GOVERNMENT IS A SLAVE MASTER'],['Metaphor_family_Governance_metaphors'],['GOVERNMENT IS A SLAVE MASTER'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Metaphor_ADDRESSING_POVERTY_IS_TREATING_AN_ILL...,['Frame_Treating_a_physical_affliction'],['Frame_Addressing_poverty'],"['Example_198', 'Example_196', 'Example_199', ...",NaN,"['Entailed', 'Composed/complex']",NaN,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],['Metaphor_family_Physical_afflictions_metapho...,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Metaphor_BEING_IMPOVERISHED_IS_BEING_AT_A_LOW_...,['Frame_Being_at_a_Low_Location'],['Frame_Impoverished'],"['Example_587', 'Example_589', 'Example_584', ...",NaN,NaN,"['Entailment_41', 'Entailment_42', 'Entailment...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],"['Metaphor_family_Poverty_metaphors', 'Metapho...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Metaphor_TREATIES_ARE_INTERPERSONAL_CONTRACTS,['Frame_Contracts'],['Frame_Treaties'],"['Example_5418', 'Example_5414', 'Example_5412...",NaN,NaN,NaN,['TREATIES ARE INTERPERSONAL CONTRACTS'],['Metaphor_family_Governance_metaphors'],['TREATIES ARE INTERPERSONAL CONTRACTS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
869,Metaphor_WEALTH_IS_LINEAR_SPATIAL_SCALE,['Frame_Scale'],['Frame_Wealth'],"['Example_5522', 'Example_5523', 'Example_5527...",NaN,NaN,NaN,['WEALTH IS LINEAR SPATIAL SCALE'],['Metaphor_family_Wealth_metaphors'],['WEALTH IS LINEAR SPATIAL SCALE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
870,Metaphor_FORMING_WORDS_IS_SHAPING,['Frame_Crafting'],['Frame_Forming_words'],"['Example_2195', 'Example_2198', 'Example_2197...",NaN,['Entailed'],NaN,['FORMING WORDS IS SHAPING'],['Metaphor_family_Conduit_Metaphor'],['FORMING WORDS IS SHAPING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
871,Metaphor_COMMUNICATING_IS_POINTING,['Frame_Cause_to_perceive'],['Frame_Communication'],"['Example_1114', 'Example_1113', 'Example_1115...",NaN,['Primary'],NaN,['COMMUNICATING IS POINTING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['COMMUNICATING IS POINTING'],...,"[""Pointing to something draws the person's att...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
872,Metaphor_INCREASING_EVALUABILITY_OF_GOVERNMENT...,['Frame_TBD'],['Frame_TBD'],"['Example_3172', 'Example_3177', 'Example_3176...",NaN,NaN,NaN,['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,['Metaphor_family_Governance_metaphors'],['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [4]:
df.hasSourceFrame.value_counts()

hasSourceFrame
['Frame_TBD']                                    62
['Frame_Verticality']                            28
['Frame_Motion_along_a_path']                    17
['Frame_Machine']                                14
['Frame_Possession']                             13
                                                 ..
['Frame_Clear_paths_to_accessing_a_location']     1
['Frame_Actor']                                   1
['Frame_Equalizing_quantities']                   1
['Frame_Caused_downward_motion']                  1
['Frame_Cause_to_perceive']                       1
Name: count, Length: 305, dtype: int64

In [6]:
df.hasTargetFrame.value_counts()

hasTargetFrame
['Frame_TBD']                          62
['Frame_Nation']                       22
['Frame_Government_institution']       17
['Frame_Taxation']                     16
['Frame_Improving_economic_status']    16
                                       ..
['Frame_Plurality']                     1
['Frame_Spatial_relation']              1
['Frame_Positivity_scale']              1
['Frame_Limited_resource']              1
['Frame_Forming_words']                 1
Name: count, Length: 261, dtype: int64

In [7]:
counts = pd.concat(
    [df["hasSourceFrame"], df["hasTargetFrame"]],
    ignore_index=True
).dropna().value_counts()

In [8]:
counts

['Frame_TBD']                        124
['Frame_Verticality']                 28
['Frame_Nation']                      24
['Frame_Government_institution']      17
['Frame_Motion_along_a_path']         17
                                    ... 
['Frame_Body_weight']                  1
['Frame_Paths_out_of_a_location']      1
['Frame_Concentration']                1
['Frame_Sentence']                     1
['Frame_Forming_words']                1
Name: count, Length: 526, dtype: int64

In [10]:
df.hasMetaphorType.value_counts()

hasMetaphorType
['Composed/complex']                           278
['Primary']                                    137
['Primary', 'Entailed']                         23
['Entailed']                                    20
['Composed/complex', 'Entailed']                17
['Entailed', 'Composed/complex']                16
['Entailed', 'Primary']                         10
['Primary', 'Composed/complex']                  5
['Composed/complex', 'Primary']                  2
['Composed/complex', 'Entailed', 'Primary']      1
['Primary', 'Entailed', 'Composed/complex']      1
Name: count, dtype: int64

There are 510 total metaphors in MetaNet. Of these, 179 of the metaphors include a label of "Primary" among the listed metaphor types. 137 metaphors are labeled exclusively as Primary. These will be assessed to confirm that the metaphor types do not cause conflict.

In [13]:
primary = df[df["hasMetaphorType"].str.contains("Primary", na=False)]

In [17]:
primary

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
6,Metaphor_ACHIEVING_A_PURPOSE_IS_REACHING_A_DES...,['Frame_Motion_to_a_location'],['Frame_Purposeful_action'],"['Example_91', 'Example_92', 'Example_93', 'Ex...",NaN,"['Primary', 'Entailed']","['Entailment_8', 'Entailment_7', 'Entailment_9']",['ACHIEVING A PURPOSE IS REACHING A DESTINATION'],"['Metaphor_family_Grady_Primary_Metaphors', 'M...",['ACHIEVING A PURPOSE IS REACHING A DESTINATION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
10,Metaphor_PROGRESS_IS_FORWARD_MOTION_ALONG_A_PATH,['Frame_Motion_along_a_path'],['Frame_Action'],"['Example_4628', 'Example_4627', 'Example_4630...",NaN,"['Primary', 'Entailed']",NaN,['PROGRESS IS FORWARD MOTION ALONG A PATH'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['PROGRESS IS FORWARD MOTION ALONG A PATH'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
850,Metaphor_MEANS_ARE_PATHS,['Frame_Self_propelled_motion_to_a_destination'],['Frame_Purposeful_action'],"['Example_3701', 'Example_3702', 'Example_3707...",NaN,"['Entailed', 'Primary']",['Entailment_165'],['MEANS ARE PATHS'],"['Metaphor_family_Grady_Primary_Metaphors', 'M...",['MEANS ARE PATHS'],...,['The correlation between goal-oriented decisi...,NaN,['Metaphor_ACHIEVING_A_PURPOSE_IS_REACHING_A_D...,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [15]:
primary.hasSourceFrame.value_counts()

hasSourceFrame
['Frame_Verticality']            20
['Frame_Object']                  7
['Frame_Motion_along_a_path']     7
['Frame_Bounded_region']          6
['Frame_Being_at_a_location']     5
                                 ..
['Frame_Aids_to_motion']          1
['Frame_Front']                   1
['Frame_Weight']                  1
['Frame_Digestion']               1
['Frame_Cause_to_perceive']       1
Name: count, Length: 95, dtype: int64

In [16]:
primary.hasTargetFrame.value_counts()

hasTargetFrame
['Frame_Time']                 8
['Frame_Action']               8
['Frame_Emotion']              7
['Frame_Quantity']             7
['Frame_Purposeful_action']    5
                              ..
['Frame_Life']                 1
['Frame_Change_of_state']      1
['Frame_Social_group']         1
['Frame_Assistance']           1
['Frame_Becoming_aware']       1
Name: count, Length: 96, dtype: int64

In [24]:
import ast

def is_primary_only(x):
    try:
        return ast.literal_eval(x) == ["Primary"]
    except Exception:
        return False

primary_only = df[df["hasMetaphorType"].apply(is_primary_only)]


In [25]:
primary_only

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BEING WEALTHY IS BEING FREE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...",['COMMUNICATING IS SHOWING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",['HAPPINESS IS VERTICALITY'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [26]:
primary_only.hasSourceFrame.value_counts()


hasSourceFrame
['Frame_Verticality']            10
['Frame_Object']                  7
['Frame_Bounded_region']          6
['Frame_Being_at_a_location']     5
['Frame_Motion_along_a_path']     4
                                 ..
['Frame_Making']                  1
['Frame_Smelly']                  1
['Frame_Fluid_motion']            1
['Frame_Measuring']               1
['Frame_Cause_to_perceive']       1
Name: count, Length: 82, dtype: int64

In [27]:
primary_only.hasTargetFrame.value_counts()


hasTargetFrame
['Frame_Time']                      6
['Frame_Action']                    4
['Frame_Causation']                 4
['Frame_Difficulties_in_action']    4
['Frame_Thinking_about']            3
                                   ..
['Frame_Change_of_state']           1
['Frame_Sympathy']                  1
['Frame_Organization']              1
['Frame_Abstract_relatedness']      1
['Frame_Becoming_aware']            1
Name: count, Length: 89, dtype: int64

In [30]:
primary_only

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
12,Metaphor_DISTINGUISHING_IS_DIVIDING,['Frame_Dividing'],['Frame_Distinguishing'],"['Example_1700', 'Example_1701', 'Example_1705...",NaN,['Primary'],NaN,['DISTINGUISHING IS DIVIDING'],['Metaphor_family_Primary_Metaphors'],['DISTINGUISHING IS DIVIDING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
18,Metaphor_ACTION_IS_MOTION,['Frame_Motion'],['Frame_Action'],"['Example_136', 'Example_134', 'Example_132', ...","[""What is the appropriate Source frame here? ...",['Primary'],NaN,['ACTION IS MOTION'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['ACTION IS MOTION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
26,Metaphor_ACQUIRING_IDEAS_IS_EATING,['Frame_Eating'],['Frame_Thinking_about'],"['Example_108', 'Example_115', 'Example_107', ...",['See the numerous subcases of this metaphor f...,['Primary'],['Entailment_10'],['ACQUIRING IDEAS IS EATING'],"['Metaphor_family_Mind_Metaphors', 'Metaphor_f...",['ACQUIRING IDEAS IS EATING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
33,Metaphor_BEING_WEALTHY_IS_BEING_FREE,['Frame_Lack_of_impediment_to_motion'],['Frame_Wealth'],"['Example_647', 'Example_649', 'Example_650', ...",NaN,['Primary'],NaN,['BEING WEALTHY IS BEING FREE'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BEING WEALTHY IS BEING FREE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
38,Metaphor_COMMUNICATING_IS_SHOWING,['Frame_Cause_to_see'],['Frame_Communication'],"['Example_1124', 'Example_1117', 'Example_1121...",NaN,['Primary'],['Entailment_66'],['COMMUNICATING IS SHOWING'],"['Metaphor_family_Conduit_Metaphor', 'Metaphor...",['COMMUNICATING IS SHOWING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
844,Metaphor_HAPPINESS_IS_VERTICALITY,['Frame_Verticality'],['Frame_Happiness'],"['Example_2821', 'Example_2822', 'Example_2818...",NaN,['Primary'],"['Entailment_123', 'Entailment_124']",['HAPPINESS IS VERTICALITY'],"['Metaphor_family_Happiness_Metaphors', 'Metap...",['HAPPINESS IS VERTICALITY'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
845,Metaphor_ATTRIBUTES_ARE_POSSESSIONS,['Frame_Possession'],['Frame_Attribute'],"['Example_523', 'Example_524', 'Example_521', ...",NaN,['Primary'],"['Entailment_38', 'Entailment_37', 'Entailment...",['ATTRIBUTES ARE POSSESSIONS'],['Metaphor_family_Object_Event_Structure_Metap...,['ATTRIBUTES ARE POSSESSIONS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
861,Metaphor_EXPERIENCE_OF_A_STATE_IS_SPATIAL_RELA...,['Frame_Spatial_relation'],['Frame_Experiencing_a_state'],"['Example_2101', 'Example_2098', 'Example_2097...",['This is a non-perspectivized view of the spa...,['Primary'],"['Entailment_93', 'Entailment_92', 'Entailment...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,"['Metaphor_family_Orientational_Metaphors', 'M...",['EXPERIENCE OF A STATE IS SPATIAL RELATION WI...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
864,Metaphor_BECOMING_AWARE_IS_NOTICING,['Frame_Noticing'],['Frame_Becoming_aware'],"['Example_555', 'Example_556', 'Example_553', ...",NaN,['Primary'],NaN,['BECOMING AWARE IS NOTICING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['BECOMING AWARE IS NOTICING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [33]:
# === 1) produce primary_frames.txt ===
import ast
import re
import pandas as pd

# If df isn't already loaded, load it:
# df = pd.read_csv("path/to/your.csv")

def is_primary_only(x):
    """Return True only if x represents exactly ['Primary'] or the string 'Primary'."""
    if pd.isna(x):
        return False
    if x == "Primary":
        return True
    if isinstance(x, list):
        return x == ["Primary"]
    if isinstance(x, str):
        # Try literal_eval for list-like strings: "['Primary']"
        try:
            val = ast.literal_eval(x)
            return val == ["Primary"]
        except Exception:
            # Fall back to delimiter-splitting (e.g. "Primary; Complex" -> not exclusive)
            parts = re.split(r'[;,]\s*', x.strip())
            parts = [p for p in parts if p]
            return len(parts) == 1 and parts[0] == "Primary"
    return False

# mask rows that are exclusively Primary
mask = df["hasMetaphorType"].apply(is_primary_only)

# collect frames that appear in source or target columns within those rows
src = df.loc[mask, "hasSourceFrame"].dropna()
tgt = df.loc[mask, "hasTargetFrame"].dropna()


def normalize_frame_name(x):
    # Case 1: actual list
    if isinstance(x, list):
        if len(x) != 1:
            return None
        x = x[0]

    # Case 2: list-like string: "['frame_action']"
    if isinstance(x, str):
        try:
            val = ast.literal_eval(x)
            if isinstance(val, list) and len(val) == 1:
                x = val[0]
        except Exception:
            pass

    if not isinstance(x, str):
        return None

    x = x.strip().lower()
    if x.startswith("frame_"):
        x = x[len("frame_"):]
    return x


primary_frames = (
    pd.concat([src, tgt])
      .map(normalize_frame_name)
      .dropna()
      .unique()
)

primary_set = set(primary_frames)

with open("data/primary_frames.txt", "w", encoding="utf-8") as fw:
    for f in sorted(primary_set):
        fw.write(f + "\n")

print(f"Wrote {len(primary_set)} frames to primary_frames.txt")


Wrote 165 frames to primary_frames.txt


In [34]:
# === 2) filter the wordlists file by the primary frames ===
import re

INPUT = "data/word_lists.txt"                   # original file (Frame_...: format)
OUTPUT = "data/word_lists_primary_only.txt"     # filtered output

frame_header_re = re.compile(r'^Frame_(.+?):\s*$')
pos_re = re.compile(r'\.[A-Za-z]+$')            # .n .v .a etc.
lead_marker_re = re.compile(r'^[\-\*\•\s]+')    # drop bullets/spaces

with open(INPUT, "r", encoding="utf-8") as fin, \
     open(OUTPUT, "w", encoding="utf-8") as fout:

    current_frame = None
    include_current = False
    words = []
    seen = set()

    for raw in fin:
        line = raw.strip()
        if not line:
            continue

        m = frame_header_re.match(line)
        if m:
            # flush previous frame if included
            if include_current and words:
                fout.write(current_frame + "\n")
                for w in words:
                    fout.write(w + "\n")
                fout.write("\n")

            current_frame = m.group(1).lower()
            include_current = current_frame in primary_set
            words = []
            seen = set()
            continue

        if not include_current:
            continue

        token = lead_marker_re.sub('', line)
        # skip multiword entries
        if " " in token:
            continue
        token = pos_re.sub('', token).strip()
        if not token:
            continue
        if token in seen:
            continue
        seen.add(token)
        words.append(token)

    # final flush
    if include_current and words:
        fout.write(current_frame + "\n")
        for w in words:
            fout.write(w + "\n")
        fout.write("\n")

print("Wrote filtered wordlists to:", OUTPUT)


Wrote filtered wordlists to: data/word_lists_primary_only.txt


In [35]:
df

,name,hasSourceFrame,hasTargetFrame,hasExample,comment,hasMetaphorType,hasEntailment,hasName,isInMetaphorFamily,label,...,hasExperientialBasis,isRelatedToMetaphorBySource,isAMappingWithinMetaphor,Metaphor_Comments,isADualOfMetaphor,isRelatedToMetaphorByTarget,hasTransitiveSubpart2Metaphor,hasTransitiveSubpart1Metaphor,Metaphor_Experientialbasis,Relatedmetaphor_Description
0,Metaphor_POWER_IS_A_POSSESSED_OBJECT,['Frame_Possession'],['Frame_Power'],"['Example_4540', 'Example_4542', 'Example_4541...",['this metaphor is similar to CONTROL IS A POS...,['Composed/complex'],"['Entailment_196', 'Entailment_195', 'Entailme...",['POWER IS A POSSESSED OBJECT'],['Metaphor_family_Possession_metaphors'],['POWER IS A POSSESSED OBJECT'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Metaphor_GOVERNMENT_IS_A_SLAVE_MASTER,['Frame_Work_scenario'],['Frame_Government'],"['Example_2625', 'Example_2621', 'Example_2626...",NaN,NaN,NaN,['GOVERNMENT IS A SLAVE MASTER'],['Metaphor_family_Governance_metaphors'],['GOVERNMENT IS A SLAVE MASTER'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Metaphor_ADDRESSING_POVERTY_IS_TREATING_AN_ILL...,['Frame_Treating_a_physical_affliction'],['Frame_Addressing_poverty'],"['Example_198', 'Example_196', 'Example_199', ...",NaN,"['Entailed', 'Composed/complex']",NaN,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],['Metaphor_family_Physical_afflictions_metapho...,['ADDRESSING POVERTY IS TREATING AN ILLNESS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Metaphor_BEING_IMPOVERISHED_IS_BEING_AT_A_LOW_...,['Frame_Being_at_a_Low_Location'],['Frame_Impoverished'],"['Example_587', 'Example_589', 'Example_584', ...",NaN,NaN,"['Entailment_41', 'Entailment_42', 'Entailment...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],"['Metaphor_family_Poverty_metaphors', 'Metapho...",['BEING IMPOVERISHED IS BEING AT A LOW LOCATION'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Metaphor_TREATIES_ARE_INTERPERSONAL_CONTRACTS,['Frame_Contracts'],['Frame_Treaties'],"['Example_5418', 'Example_5414', 'Example_5412...",NaN,NaN,NaN,['TREATIES ARE INTERPERSONAL CONTRACTS'],['Metaphor_family_Governance_metaphors'],['TREATIES ARE INTERPERSONAL CONTRACTS'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
869,Metaphor_WEALTH_IS_LINEAR_SPATIAL_SCALE,['Frame_Scale'],['Frame_Wealth'],"['Example_5522', 'Example_5523', 'Example_5527...",NaN,NaN,NaN,['WEALTH IS LINEAR SPATIAL SCALE'],['Metaphor_family_Wealth_metaphors'],['WEALTH IS LINEAR SPATIAL SCALE'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
870,Metaphor_FORMING_WORDS_IS_SHAPING,['Frame_Crafting'],['Frame_Forming_words'],"['Example_2195', 'Example_2198', 'Example_2197...",NaN,['Entailed'],NaN,['FORMING WORDS IS SHAPING'],['Metaphor_family_Conduit_Metaphor'],['FORMING WORDS IS SHAPING'],...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
871,Metaphor_COMMUNICATING_IS_POINTING,['Frame_Cause_to_perceive'],['Frame_Communication'],"['Example_1114', 'Example_1113', 'Example_1115...",NaN,['Primary'],NaN,['COMMUNICATING IS POINTING'],"['Metaphor_family_Primary_Metaphors', 'Metapho...",['COMMUNICATING IS POINTING'],...,"[""Pointing to something draws the person's att...",NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
872,Metaphor_INCREASING_EVALUABILITY_OF_GOVERNMENT...,['Frame_TBD'],['Frame_TBD'],"['Example_3172', 'Example_3177', 'Example_3176...",NaN,NaN,NaN,['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,['Metaphor_family_Governance_metaphors'],['INCREASING EVALUABILITY OF GOVERNMENT IS INC...,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
